Iter9

MAGeorg opened the pull request.

Commit:
iter6: add logger

Commit:
iter7: add new endpoint

Commit:
iter8: add compressed

Commit:
iter9: add worked with file

arttet commented on cmd/shortener/main.go:
```diff
@@ -12,14 +13,48 @@ func main() {
+	// проверка конфига на наличие переменной для хранения имени файла
+	var lastID int
+	var err error
+	if cfg.StorageFileName != "" {
```
CHANGE REQUEST: please simplify it

arttet commented on internal/config/config.go:
```diff
@@ -6,22 +6,29 @@ import (
+	flag.StringVar(&conf.StorageFileName, "f", "/tmp/short-url-db.json", "Full path to the file where the created shortened URLs are stored")
 	flag.Parse()
+
+	if a := os.Getenv("SERVER_ADDRESS"); a != "" {
```
CHANGE REQUEST: this is a magic name

arttet commented on internal/handlers/handlers.go:
```diff
@@ -38,6 +40,15 @@ func (h *AppHandler) CreateHashURL(w http.ResponseWriter, r *http.Request) {
+
+	if h.a.LastID != -1 {
+		h.a.LastID += 1
+		err := h.a.Producer.WriteEvent(&models.Event{ID: h.a.LastID, HashURL: hash, URL: string(urlStr)})
```
CHANGE REQUEST: we shouldn't do it here

arttet commented on internal/logger/logger.go:
```diff
@@ -0,0 +1,28 @@
+	"go.uber.org/zap"
+)
+
+var Sugar zap.SugaredLogger
```
CHANGE REQUEST: please don't use global variables

arttet commented on internal/middleware/gzip.go:
```diff
@@ -0,0 +1,111 @@
+	return false
+}
+
+func MiddlewareGzip(h http.HandlerFunc) http.HandlerFunc {
```
```suggestion
func GzipMiddleware(h http.HandlerFunc) http.HandlerFunc {
```

arttet commented on internal/models/models.go:
```diff
@@ -0,0 +1,20 @@
+
+// структура для обработки ответа POST запроса, содержащего
+// сокращенный URL в теле в формате JSON
+type AnswerHashURL struct {
```
```suggestion
type ResponseHashURL struct {
```

Commit:
iter9: corrections in work

1. corrected gitignore so that it does not include test and server binaries in Git;
2. names of structures and functions have been changed;
3. magic name is made into constants;
4. removed global logger variable;
5. business logic has been removed from the handler code.

MAGeorg commented:
golangci-lint и covarege добавлен в первом sprint review
могу ли я дополнительное задание в виде swagger, Dependabot в процессе выполнения sprint 3, если вдруг не успею до окончания приема 2ого sprit?

Commit:
iter9: added additional tasks

1. Added openapi specifications
2. Added config for Dependabot
3. Changed response statuses for endpoints

Commit:
iter9: fixed test after changed status code

Commit:
iter9: corrections in work

business logic is taken out of handlers

arttet commented on internal/core/stor_handler.go:
```diff
@@ -0,0 +1,41 @@
+}
+
+// функция реализует бизнес логику обработки начального URL
+func CreateShotURL(i InputValueForWriteFile) (string, error) {
```
```suggestion
func CreateShotURL(i *InputValueForWriteFile) (string, error) {
```

arttet commented on internal/core/stor_handler.go:
```diff
@@ -0,0 +1,41 @@
+	if err != nil {
+		return "", fmt.Errorf("error add url to storage")
+	}
+	if *i.LastID != -1 {
```
This is not an efficient solution.
